### ISSUE TITLE
disableFeedback breaks the custom persistence layer

### ISSUE THREAD

**Issue by smannathan** [Chainlit/chainlit#1179]
An error occurred: (sqlalchemy.dialects.postgresql.asyncpg.IntegrityError) <class 'asyncpg.exceptions.NotNullViolationError'>: null value in column "disableFeedback" of relation "steps" violates not-null constraint

CREATE TABLE IF NOT EXISTS steps (
    "id" UUID PRIMARY KEY,
    "name" TEXT NOT NULL,
    "type" TEXT NOT NULL,
    "threadId" UUID NOT NULL,
    "parentId" UUID,
    **"disableFeedback" BOOLEAN NOT NULL,** -> Updated this field to NULLABLE to fix this issue
    "streaming" BOOLEAN NOT NULL,
    "waitForAnswer" BOOLEAN,
    "isError" BOOLEAN,
    "metadata" JSONB,
    "tags" TEXT[],
    "input" TEXT,
    "output" TEXT,
    "createdAt" TEXT,
    "start" TEXT,
    "end" TEXT,
    "generation" JSONB,
    "showInput" TEXT,
    "language" TEXT,
    "indent" INT
);

**Comment by nvinayvarma189** [comment]
Facing the same issue with the latest version of chainlit

At the time of writing, this is how the steps table is defined [here](https://docs.chainlit.io/data-persistence/custom)
```
CREATE TABLE IF NOT EXISTS steps (
    "id" UUID PRIMARY KEY,
    "name" TEXT NOT NULL,
    "type" TEXT NOT NULL,
    "threadId" UUID NOT NULL,
    "parentId" UUID,
    "disableFeedback" BOOLEAN NOT NULL,
    "streaming" BOOLEAN NOT NULL,
    "waitForAnswer" BOOLEAN,
    "isError" BOOLEAN,
    "metadata" JSONB,
    "tags" TEXT[],
    "input" TEXT,
    "output" TEXT,
    "createdAt" TEXT,
    "start" TEXT,
    "end" TEXT,
    "generation" JSONB,
    "showInput" TEXT,
    "language" TEXT,
    "indent" INT
);
```

This is not in sync with the StepDict defined in the library code [here](https://github.com/Chainlit/chainlit/blob/643585629e437621b037ed3a3da428bc33b12dc0/backend/chainlit/step.py#L45). The `disableFeedback` attribute is missing here and instead has a `feedback` column

**Comment by AidanShipperley** [comment]
Yes, the community hasn't updated the documentation to reflect this change.

You need to update your `Steps` table's schema to allow the `disableFeedback` column to be nullable, or you need to delete the `disableFeedback` column from your `Steps` table completely. This column is no longer used anywhere as far as I am aware.

Additionally, if you are using a custom data layer, you need to update your data layer's Python code to remove the collection of the `disableFeedback` column in the `get_all_user_threads()` function, as shown [here](https://github.com/Chainlit/chainlit/pull/1128/files#diff-8cd91330000b6a390840c3da1b47faf723dab4e195d45694fb6e46751ca1d7af).

In reference to @nvinayvarma189's comment, there is no `feedback` column you need to add to your `Steps` table, this was always in the StepDict class. This value is constructed in [`get_all_user_threads()`](https://github.com/Chainlit/chainlit/blob/643585629e437621b037ed3a3da428bc33b12dc0/backend/chainlit/data/sql_alchemy.py#L558-L563), where the feedback value is collected through [joining the Steps table to the Feedbacks table](https://github.com/Chainlit/chainlit/blob/643585629e437621b037ed3a3da428bc33b12dc0/backend/chainlit/data/sql_alchemy.py#L485-L510).

**Comment by Physium** [comment]
whats so hard about making this simple change in the docs? errors like this causes a ton of unnecessary troubleshooting.
